From 86a5f84c495d72f6483fd3978342415def866538 Mon Sep 17 00:00:00 2001
From: wang-bin <wbsecg1@gmail.com>
Date: Sun, 23 Jul 2023 23:58:24 +0800
Subject: [PATCH 33/33] vulkan: reduce insane array size

if array length is AV_CODEC_ID_FIRST_AUDIO, then size is 16^4*16=1MB, it's a huge stack size.
better to use a function with switch/case.
---
 libavcodec/vulkan_video.c | 2 +-
 libavcodec/vulkan_video.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/libavcodec/vulkan_video.c b/libavcodec/vulkan_video.c
index 460ccf6ca8..184afd5508 100644
--- a/libavcodec/vulkan_video.c
+++ b/libavcodec/vulkan_video.c
@@ -20,7 +20,7 @@
 
 #include "vulkan_video.h"
 
-const FFVkCodecMap ff_vk_codec_map[AV_CODEC_ID_FIRST_AUDIO] = {
+const FFVkCodecMap ff_vk_codec_map[] = {
     [AV_CODEC_ID_H264] = {
                            0,
                            0,
diff --git a/libavcodec/vulkan_video.h b/libavcodec/vulkan_video.h
index 183ce89bf0..13237a630c 100644
--- a/libavcodec/vulkan_video.h
+++ b/libavcodec/vulkan_video.h
@@ -49,7 +49,7 @@ typedef struct FFVkVideoSession {
 /**
  * Index is codec_id.
  */
-extern const FFVkCodecMap ff_vk_codec_map[AV_CODEC_ID_FIRST_AUDIO];
+extern const FFVkCodecMap ff_vk_codec_map[];
 
 /**
  * Get pixfmt from a Vulkan format.
-- 
2.39.2 (Apple Git-143)

